src/gasPriceOracle/oracle.ts

```diff
@@ -17,24 +21,46 @@ import * as polygon from "./adapters/polygon";
 export async function getGasPriceEstimate(
   provider: providers.Provider,
   chainId?: number,
+  transport?: Transport,
   legacyFallback = true
 ): Promise<GasPriceEstimate> {
   if (chainId === undefined) {
     ({ chainId } = await provider.getNetwork());
   }
 
-  const gasPriceFeeds: { [chainId: number]: GasPriceFeed } = {
+  const useViem = process.env[`NEW_GAS_PRICE_ORACLE_${chainId}`] === "true";
```

Comment:
I generally think reading env-vars directly in application code is an anti-pattern and I would normally propose config options to be implemented as part of the interface. I embedded it here because this config is only really intended as a short-term handle to support proving of viem, and I think it'd be removed as part of a cleanup task to remove ethers altogether. In that case I figured it made more sense to embed it directly to avoid imposing config burden on the calling code. wdyt?

Reply:
ok yea in that case it makes sense to me like that